tests/block_header_test.go

```diff
@@ -425,3 +439,93 @@ func getWitnessProofSteps(client *ethclient.Client, blockNumber int) ([]common.H
 	}
 	return res, nil
 }
+
+func testHeaders_LogsReferenceTheirContext(t *testing.T, headers []*types.Header, client *ethclient.Client) {
+	require := require.New(t)
+
+	numLogs := 0
+	for _, header := range headers {
+		receipts, err := client.BlockReceipts(context.Background(),
+			rpc.BlockNumberOrHashWithHash(header.Hash(), false))
+		require.NoError(err, "failed to get block receipts")
+
+		blockHash := header.Hash()
+		blockNumber := header.Number.Uint64()
```

Comment:
why not do these declarations before getting the receipt and using the hash variable in `rpc.BlockNumberOrHashWithHash` ?